File: src/main/java/com/googongill/aditory/service/dto/category/CategoryInfo.java
package com.googongill.aditory.service.dto.category;

import com.googongill.aditory.domain.enums.CategoryState;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Builder
public class CategoryInfo {
    private Long categoryId;
    private String categoryName;
    private String asCategoryName;
    private Integer linkCount;
    private Integer likeCount;
    private CategoryState categoryState;
    @Builder.Default
    private List<String> prevLinks = new ArrayList<>();
    private LocalDateTime createdAt;
    private LocalDateTime lastModifiedAt;
}

File: src/main/java/com/googongill/aditory/service/dto/category/CategoryInfo.kt
package com.googongill.aditory.service.dto.category

import com.googongill.aditory.domain.enums.CategoryState
import java.time.LocalDateTime

data class CategoryInfo(
    val categoryId: Long?,
    val categoryName: String,
    val asCategoryName: String,
    val linkCount: Int,
    val likeCount: Int,
    val categoryState: CategoryState,
    val prevLinks: List<String> = emptyList(),
    val createdAt: LocalDateTime?,
    val lastModifiedAt: LocalDateTime?
)
